RevenueCatUI/Purchasing/PaywallPurchasesType.swift

```diff
@@ -17,12 +17,17 @@ import RevenueCat
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 protocol PaywallPurchasesType: Sendable {
 
+    var finishTransactions: Bool { get set }
```

Comment:
I find this name a bit ambiguous, and personally I'd want to read the docs to make sure I understood what it meant when it was true or false. Additionally Joop pointed out that it has the opposite logic of Observer Mode, which makes it more likely customers who are migrating from one to the other may not negate the value they are passing, resulting in bugs. Cesar also noted that the concept of "transactions" is iOS-specific.

I have not thought of a perfect alternative, but maybe `customPurchaseImplementation` or `externalPurchaseImplementation`?

Thoughts?

Reply:
https://revenuecat.slack.com/archives/C075CNL6YAV/p1717057280915299

Reply:
I've updated it to `purchasesAreCompletedBy` of enum type `PurchasesAreCompletedBy`, and deprecated the publicly facing instances of `finishTransactions`. `finishTransactions` is still used internally, unsure if I should update those too....

Reply:
@jamesrb1 Should we change this base branch from `main` to `5.0-dev` if we including this enum change in there since its a breaking change? 🤔

Reply:
I don't think this should break anything, but you will get deprecation warnings if you use the `.finishTransactions` property on `Purchases`. The config builder still refers to Observer Mode - is that change to be part of this PR too....?

Reply:
Just FYI: we merged to `main` in purchases-android, since it's not breaking indeed, just deprecating.